db/toblock.c

```diff
@@ -2400,7 +2400,9 @@ static void backout_and_abort_tranddl(struct ireq *iq, tran_type *parent,
     int rc = 0;
     if (rowlocks) {
         assert(parent);
-        rc = trans_abort(iq, bdb_get_physical_tran(parent));
+        /* Check if we started a physical transaction. */
+        if (bdb_get_physical_tran(parent) != NULL)
+            rc = trans_abort(iq, bdb_get_physical_tran(parent));
         bdb_reset_physical_tran(parent);
```

Comment:
The call to bdb_reset_physical_tran() should be inside the "if" block here.

Reply:
Thank you, Joe! I've moved the lines inside the if block.